src/components/App/App.js

```diff
@@ -24,11 +25,44 @@ type Props = {
   selectedProjectId: ?Project,
 };
 
-class App extends PureComponent<Props> {
+type State = {
+  isNodeJsMissing: boolean,
+};
+
+const { dialog, shell } = remote;
+
+class App extends PureComponent<Props, State> {
+  state = {
+    isNodeJsMissing: false,
+  };
```

Comment:
Overall, I like how you've solved this problem, using React state... but I feel a bit nervous that it's in `App.js`. Maybe I'm over-reacting, but I've worked on too many codebases where the `App.js` gets super long and complicated because it absorbs a bunch of concerns like this.

Mostly this is my fault, as I've already been using App for analytics and event-listening. But what do you think of updating it so that we have a new component that manages this:

```js
class Initialization extends Component {
  ... all the lifecycles and stuff

  render() {
    return this.props.children(this.state.wasSuccessfullyInitialized);
  }
}

// So that in App's render, we can do:

return (
  <Initialization>
    {wasSuccessfullyInitialized => (
      ... all the stuff currently in App render
    )}
  </Initialization>
)
```

The benefit IMO is that when folks are just trying to see how the app is structured, they can check `App.js` and just see what components it renders, without getting lost in a bunch of analytics and edge-case details.

Reply:
Also, in that example, I changed `isNodeJsMissing` to `wasSuccessfullyInitialized`, for two reasons:

1. It feels like NodeJS being available is an implementation detail for `Initialization`, and not something `App` needs to know about. It just needs to know if everything worked or not.

2. I inverted the logic, so that the good condition is `true` instead of `false`. In general I prefer this approach, because I find it simpler to check for truthiness (no double-negative of `!wasNotInitialized`), and because we can set the initial value to `false` (in your current example, I believe that the `CreateNewProjectWizard` would mount on first render regardless of whether or not the user has NodeJS installed. It's not a big deal since that component isn't visible anyway, but this way we avoid the cost of mounting it right away)

Reply:
Sounds great. I'll create the Initialization component & you're right with the naming.